eng/pipelines/templates/stages/archetype-java-release.yml

```diff
@@ -130,7 +130,7 @@ stages:
                         ArtifactLocation: $(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed/${{artifact.groupId}}/${{artifact.name}}
                         PackageRepository: Maven
                         ReleaseSha: $(Build.SourceVersion)
-
+                    - template: /eng/common/pipelines/templates/steps/retain-run.yml
```

Comment:
It seems as though you are hooking this up after the tag template consistently. Do you think it is worth just adding it to the end of the template instead?

Reply:
I see it as a discrete step to tagging, it just so happens that tagging in these pipelines is the first change outside of the pipeline that occurs as part of release. If we have pipelines that have something that happens before tagging, it should be moved there instead.